### PR TITLE
fix-data-packed-with-wrong-version-of-caniuse-db

### PR DESCRIPTION
**Pull request by Beretta1979**
This fixes issue #42 

I found out that the exec steps in the publish script where not waited upon by the next steps. This can cause next steps to use the wrong version of the `caniuse-db` package as the yarn publish is not finished when starting the pack.
I changed the publish script so that child processes are properly handled. This is based on work done in the https://github.com/sindresorhus/np repo, where some parts of the publish script were originally based upon.

### CODE REVIEW

File: publish.js [hunk 1]
```diff
@@ -11,9 +12,9 @@ const remark = require('remark');
 const u = require('unist-builder');
 const fs = require('mz/fs');
 const fecha = require('fecha');
-require('any-observable/register/rxjs-all');
-const Observable = require('any-observable');
-const streamToObservable = require('stream-to-observable');
+const {merge, throwError} = require('rxjs');
+const {catchError, filter, finalize} = require('rxjs/operators');
+const streamToObservable = require('@samverschueren/stream-to-observable');
```
Comment (ai) [inline publish.js]: What is the difference between these implementations of `streamToObservable`?

File: publish.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require('any-observable/register/rxjs-all');
```
Comment (ai) [inline publish.js]: Why do we need this change?

File: publish.js [hunk 3]
```diff
@@ -11,9 +12,9 @@ const remark = require('remark');
 const u = require('unist-builder');
 const fs = require('mz/fs');
 const fecha = require('fecha');
-require('any-observable/register/rxjs-all');
-const Observable = require('any-observable');
-const streamToObservable = require('stream-to-observable');
+const {merge, throwError} = require('rxjs');
+const {catchError, filter, finalize} = require('rxjs/operators');
```
Comment (ai) [inline publish.js]: Why I do not see `catchError` and `finalize` usage above?

Comment (Beretta1979) [reply publish.js]: Removed them

File: publish.js (hunk 2 above)
Comment (Beretta1979) [reply publish.js]: This change is indeed not needed

File: publish.js (hunk 1 above)
Comment (Beretta1979) [reply publish.js]: The listr package is used as the publish task runner. `@samverschueren/stream-to-observable` is already a dependency of that package. It seems to be more maintained than the forked package. Not sure if it is needed though.

**Comment by ai** [comment]
Looks good. I will just finish all my current tasks for Browserslist and PostCSS and then will double-check the PR and update the auto-release machine. It could take a day or a few.

**Comment by ai** [comment]
I hope I updated the auto-release script right.

**Pull request merged**